src/main/java/com/aodocs/partialresponse/servlet/PartialResponseEndpointsServlet.java

```diff
@@ -174,12 +174,23 @@ protected ObjectWriter configureWriter(ObjectWriter objectWriter) {
 							throw new BadRequestException("Invalid field selection '" + fieldsParameterValue + "'", "invalidParameter", "global");
 						}
 					}
-					RetainedFieldCheckerImpl fieldsChecker = new RetainedFieldCheckerImpl(fieldsExpression);
-					request.setAttribute(RetainedFieldChecker.REQUEST_ATTRIBUTE_NAME, fieldsChecker);
+					RequestedFieldsImpl fieldsChecker = new RequestedFieldsImpl(fieldsExpression);
+					request.setAttribute(RequestedFields.class.getName(), fieldsChecker);
```

Comment:
I think this is a bit dangerous to use the class name in stead of a public constant: a refactoring could easily change the request attribute contract which could break client code that may directly retrieve it without using the static method helper (nor the class name).